### PR TITLE
Added customers page and many other changes

### PR DESCRIPTION
**Pull request by imsamtar**
### New Changes

- using `formatjs` wrapper `svelte-i18n`
- added customers page and table
- may other changes

### CODE REVIEW

**Comment by imsamtar** [comment]
Here is what I am looking for next...

- From which Graphql API to fetch customers?

**Comment by jozef-slezak** [comment]
You can find graphql codegen config in original
https://github.com/iteria-app/example-material-ui

Dňa pi 9. 4. 2021, 10:51 SAMEER TARIQ ***@***.***> napísal(a):

**Comment by imsamtar** [comment]
@jozef-slezak Thanks

[src/views/customer/CustomerListView/UpdateCustomer.tsx Line 5](https://github.com/iteria-app/example-material-ui/blob/main/src/views/customer/CustomerListView/UpdateCustomer.tsx#L5) is importing `src/generated/graphql` where that file does not exist

If I can get graphql endpoint and query to fetch customers that would be awesome.

**Comment by jozef-slezak** [comment]
Yes, see the package.json scripts that are generating this file.

File: src/lib/Drawer.svelte [hunk 1]
```diff
@@ -14,16 +16,34 @@
...
 		}
 	];
 
+	function itemClicked(index: number) {
+		const item = items[index];
+		if (item.submenu) {
+			items[index].expended = !item.expended;
+		}
+		if (item.href) {
+			goto(item.href);
```
Comment (jozef-slezak) [inline src/lib/Drawer.svelte]: please, try to use `<ListItem link={item.href}`

File: src/lib/customers/Customers.svelte [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+	}
+
+	function getDate(date: Date) {
+		const _date = date.getDate().toString().padStart(2, '0');
+		const _month = date.getMonth().toString().padStart(2, '0');
+		const _year = date.getFullYear().toString();
+		return `${_date}/${_month}/${_year}`;
+	}
+
+	function getAddress(address: any) {
```
Comment (jozef-slezak) [inline src/lib/customers/Customers.svelte]: no need to have this function, please inline the code

File: src/lib/customers/Customers.svelte [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+
+	function selectAll() {
+		if (allSelected) {
+			customers = customers.map((c) => ({ ...c, selected: true }));
+		} else {
+			customers = customers.map((c) => ({ ...c, selected: false }));
+		}
+	}
+
+	function getDate(date: Date) {
```
Comment (jozef-slezak) [inline src/lib/customers/Customers.svelte]: please format the date using svelte formatjs like this:
```
import { date  } from `svelte-i18n`

date(new Date(customer.createdAt)) // and make the code nullsafe: of course:
// (createdAt could be null or undifned... new Date(null) creates today I believe;
// default value should be empty string if the date is )
```

**Comment by jozef-slezak** [comment]
You can find more info about graphql here:
https://github.com/iteria-app/example-material-ui/commit/55c3ca02bbab4bcbfdf7d2d42587076fd07b474a#diff-a68b5257a543815145882d363128eafe4c4a822cb3945a005053048dc372fb45R2

Please have a look for few code review comments. Keep up the good work. I am looking forward to few more commits and your are done with this issue.

**Comment by imsamtar** [comment]
Thanks

By the way In hasura db avatarUrl for customers is null.

**Comment by imsamtar** [comment]
`urql` (npm package) requires react and is not working in svelte.

There is another package @urql/svelte but it has got some issues as well.
- It throws this errors like this `can not ___ outside component initialization` 

My question is can I use some alternative to urql for browser built in fetch to query customers?

File: src/lib/Drawer.svelte (hunk 1 above)
Comment (imsamtar) [reply src/lib/Drawer.svelte]: On ListItem Component

- link prop is not used for navigation
- Here is proof [ListItem Component Source Code](https://github.com/TheComputerM/svelte-materialify/blob/master/packages/svelte-materialify/src/components/List/ListItem.svelte)
- Probably this prop is used just for styling

**Comment by jozef-slezak** [comment]
That ertor is fine. Just initialize the urql/svelte inside App.svelte
(because it uses context)

Dňa pi 9. 4. 2021, 14:22 SAMEER TARIQ ***@***.***> napísal(a):

**Comment by imsamtar** [comment]
@jozef-slezak 

there is no App.svelte file in sveltekit

So I tried it many times in $layout.svelte and index.svelte but no success

**Comment by imsamtar** [comment]
Made new pull request #5

**Comment by jozef-slezak** [comment]
I see, in that case initialize indide customers page

Dňa pi 9. 4. 2021, 15:32 SAMEER TARIQ ***@***.***> napísal(a):

**Pull request merged**